src/App.tsx

```diff
@@ -3,12 +1,42 @@
+import { FC, useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { CompletedFilter } from './types/CompletedFilter';
+import { getFilteredTodos } from './utils/getFilteredTodos';
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
-export const App: React.FC = () => {
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
```

Comment:
You are initializing the state `selectedTodo` with `null`. This could potentially lead to problems if you try to access properties of `selectedTodo` elsewhere in your code without first checking if it is not null. Consider initializing it with a default Todo object or ensure to always check for null before accessing its properties.